### ISSUE TITLE
Mix redis2_query with empty_gif module

### ISSUE THREAD

**Issue by kilgor** [openresty/redis2-nginx-module#29]
Hi,

I wanna implement simple counter bassed on redis

location = /counter.gif {
redis2_pass redis_backend;
redis2_query incr counter;
access_log off;
empty_gif;
add_header Cache-Control "max-age=0, no-cache, no-store";
}

The main problem is that redis2_query return the value into nginx buffer and broke the output. Is there any posibility to mix this together? Maybe redis_silent_query would be good if you wanna implement?

**Comment by agentzh** [comment]
@kilgor It's a common mistake. In a single location, only one nginx module is allowed to run as the content handler to generate the response. Here you have two modules ngx_redis2 and ngx_empty_gif in a single location for generating a response, which does not make sense at all and only one of them wins. If you want to combine two nginx modules running at the same content phase, you should consider using subrequests instead.

**Comment by kilgor** [comment]
@agentzh. Thanks for explanation. now it's quite clear. But we can mix add_header or headers-more-nginx-module with other "response" module. How it's is made? It work together, right?

And the second question 

location = /test.php {
default_type text/plain;
access_log off;
redis2_query incr test;
redis2_pass redis_backend;
}

In browser i get the counter response. I'm trying make this information private so maybe redis_silent_query i would be the key?

**Comment by agentzh** [comment]
@kilgor `add_header` does not work as a "content handler", that is, it does not generate a response from scratch, rather, it runs in the "output header filter" phase which just *patches* existing responses.

To fully understand the nginx processing phases and how different nginx modules can be combined, I suggest you check out the nginx tutorials below to save both your time and mine:

http://openresty.org/#eBooks

Regarding your requirement, just as I've mentioned, you really need to use NGINX subrequests there. They are also explained in the nginx tutorials mentioned above. More specifically, just check out

https://github.com/openresty/nginx-eval-module#readme

https://github.com/openresty/lua-nginx-module#ngxlocationcapture

https://github.com/openresty/lua-nginx-module#access_by_lua

for more details.

**Comment by agentzh** [comment]
@kilgor No, you don't really need a `redis2_slient_query` directive here. Otherwise you'll quickly want a `xxx_silent_xxx` variant for literally every existing directive of nginx content handler modules (like `proxy_silent_pass`, `fastcgi_silent_pass`, `echo_silent`, you name it :)).
